core/state/statedb.go

```diff
@@ -833,7 +836,8 @@ func (s *StateDB) Copy() *StateDB {
 		accountExtraDataTrie: s.db.CopyTrie(s.accountExtraDataTrie),
 	}
 
-	s.mutex.Lock()
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
```

Comment:
don't defer over here, subsequent sections like s.preimages might not be locked during writes, we get a performance hit if so. 

we will risk accept the possibility of the rest running into the same problem, e.g. preimages, snaps, accesslist